### ISSUE TITLE
Using 'type' parameter of /events/restSearch only filters at event level, not attribute level

### ISSUE THREAD

**Issue by Wachizungu** [MISP/MISP#5600]
### Work environment

| Questions                 | Answers
|---------------------------|--------------------
| Type of issue             | Bug
| OS version (server)       | Ubuntu 18.04.3 LTS
| OS version (client)       | Win10 / MISP REST Client
| PHP version               | 7.4.2
| MISP version / git hash   | 2.4.120, 7960ce4ade4e580755a4bf7d8625f97b23e7eb66
| Browser                   | N/A

### Expected behavior
When calling /events/restSearch with type parameter set, only events that have attributes of this type are selected, attributes that are not of the selected types are not included in the output.

---It's a bit of a difficult situation here because I'm not sure how this parameter should be treated in all cases... what happens if you exclude types for example? For me it would make more sense to return the event, but remove the attributes of the type. There might be a use case for excluding  events with certain attribute types that I'm not seeing though.

### Actual behavior
When calling /events/restSearch with type parameter set, only events that have attributes of this type are selected, the full event is returned and attributes that are not of the selected types are also included in the output.

### Steps to reproduce the behavior
Create a dummy event with two different types of attributes. Let's call them <attribute_type1> and <attribute_type2>. Call /events/restSearch with the following body:

{
    "returnFormat": "json",
    "eventid": <eventid>,
    "type": "<attribute_type1>"
}

Attributes with type <attribute_type2> should be excluded, but they aren't.

### Logs, screenshots, configuration dump, ...
![example-events-restsearch-type-parameter](https://user-images.githubusercontent.com/9868873/73916464-c509e400-48bd-11ea-8da8-9b2b0a29f885.PNG)

**Comment by iglocska** [comment]
Yeah agreed, I like your suggestion of using the type inclusive filter to
filter events, but using the exclusive to remove types.
